server/src/main/java/org/candlepin/resteasy/JsonProvider.java

```diff
@@ -174,13 +182,17 @@ private void addMixInAnnotationsForDTOs(ObjectMapper mapper) {
         mapper.addMixIn(GuestIdDTOArrayElement.class, DynamicPropertyFilterMixIn.class);
         mapper.addMixIn(NestedOwnerDTO.class, DynamicPropertyFilterMixIn.class);
         mapper.addMixIn(PermissionBlueprintDTO.class, DynamicPropertyFilterMixIn.class);
+        mapper.addMixIn(ProductContentDTO.class, DynamicPropertyFilterMixIn.class);
+        mapper.addMixIn(ProductDTO.class, DynamicPropertyFilterMixIn.class);
         mapper.addMixIn(RoleDTO.class, DynamicPropertyFilterMixIn.class);
         mapper.addMixIn(StatusDTO.class, DynamicPropertyFilterMixIn.class);
         mapper.addMixIn(TimestampedEntity.class, DynamicPropertyFilterMixIn.class);
         mapper.addMixIn(UserDTO.class, DynamicPropertyFilterMixIn.class);
         mapper.addMixIn(OwnerDTO.class, DynamicPropertyFilterMixIn.class);
+        mapper.addMixIn(SubscriptionDTO.class, DynamicPropertyFilterMixIn.class);
         mapper.addMixIn(UpstreamConsumerDTO.class, DynamicPropertyFilterMixIn.class);
         //TODO: Add more mix-ins here as more DTOs are ported to openapi spec
+
+        mapper.addMixIn(ProductDTO.class, ProductAttributesMixIn.class);
```

Comment:
It looks like adding a second mixin for the same target class doesn't work (it overrides the first one). So the filtering that is provided by the DynamicPropertyFilterMixIn addition doesn't work right not (I've created the following spec test to test dynamic filtering: http://pastebin.test.redhat.com/857369).

To fix it, I tried making ProductAttributesMixIn implement DynamicPropertyFilterMixIn, but that didn't work. Then I found [this](https://github.com/FasterXML/jackson-docs/wiki/JacksonMixInAnnotations#usage-notes), which says that it should work using inheritance, so I turned DynamicPropertyFilterMixIn from an interface to an abstract class, and had ProductAttributesMixIn extend it, and it did work, so we should do that